spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceFilter.java

```diff
@@ -188,55 +177,31 @@ private void publish(ApplicationEvent event) {
 		}
 	}
 
-	// TODO: move annotation keys to constants
 	protected void addRequestAnnotations(HttpServletRequest request) {
 		String uri = this.urlPathHelper.getPathWithinApplication(request);
-		this.traceManager.addAnnotation("/http/request/uri",
-				request.getRequestURL().toString());
-		this.traceManager.addAnnotation("/http/request/endpoint", uri);
-		this.traceManager.addAnnotation("/http/request/method", request.getMethod());
-
-		Enumeration<String> headerNames = request.getHeaderNames();
-		while (headerNames.hasMoreElements()) {
-			String name = headerNames.nextElement();
-			Enumeration<String> values = request.getHeaders(name);
-			while (values.hasMoreElements()) {
-				String value = values.nextElement();
-				String key = "/http/request/headers/" + name.toLowerCase();
-				this.traceManager.addAnnotation(key, value);
 
-			}
-		}
+		this.traceManager.addAnnotation(HeaderKeys.HTTP_URL, request.getRequestURL().toString());
```

Comment:
note: if you didn't extract the constant class, it would be easier to tell what is a change and what isn't. As it is, we need to scroll up and down (and look at formatting changes) to tell if something changed or not.